File: encode_base64.js
#!/usr/bin/env node
/*
 * Description: encodes a string as base64.
 * Author: E. Chris Pedro
 * Version: 2018-01-16
 */

if (process.argv.length !== 3)
{
  console.error('usage: encode_bas64.js <string>');
  process.exit(1);
}

console.log(new Buffer.from(process.argv[2]).toString('base64'));

File: encode_base64.py
#!/usr/bin/env python
###############################################################################
# Description: encodes a string as base64.
# Author: E. Chris Pedro
# Version: 2018-02-06
###############################################################################

import sys, base64

def main(argv):
    if len(sys.argv) != 2:
        print 'usage: encode_bas64.py <string>'
        sys.exit(1)
    
    print str(base64.encodestring('%s' % (sys.argv[1]))[:-1])

if __name__ == '__main__':
    main(sys.argv)

